### DISCUSSION TITLE
Why is the mode option value injected as process.env.NODE_ENV value?

### DISCUSSION THREAD

**Discussion by juunzzi** [webpack/webpack#16201] [Q&A]
### scripts

```json
  "scripts": {
    "start": "cross-env NODE_ENV=local webpack serve --mode=development",
    "start:dev": "cross-env NODE_ENV=development webpack serve  --mode=development",
    "build": " cross-env NODE_ENV=production webpack --progress --mode=production",
    "build:local": " cross-env NODE_ENV=local webpack --progress --mode=development",
    "build:dev": "cross-env NODE_ENV=development webpack --progress --mode=development",
    "storybook": "start-storybook -p 6006",
    "build-storybook": "build-storybook"
  },
```

In this way, I am injecting the process.env.NODE_ENV value into the environment where the webpack command is executed.

In my React app code, I want to look up the value injected by scripts with process.env.NODE_ENV, but in this case, the value entered in the --mode option is injected.

**In my yarn run start case** , process.env.NODE_ENV of React App is `development`.. but I want process.env.NODE_ENV to be `local` .

I'd like to see the value I inject into process.env.NODE_ENV .

**Why is the mode option value injected as process.env.NODE_ENV value?**

**Also, where is the code to inject the mode option value into process.env.NODE_ENV?**

**Comment by alexander-akait** [comment]
That described in docs, here code for this https://github.com/webpack/webpack/blob/main/lib/WebpackOptionsApply.js#L541, you can override it using the `nodeEnv` option and docs https://webpack.js.org/configuration/optimization/#optimizationnodeenv
